src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -102,6 +103,32 @@ inline __device__ __host__ int8_t cast_to(const int32_t& val)
     return static_cast<int8_t>(val & 0xff);
 }
 
+inline __device__ __host__ bool IsZero(double val) { return fabs(val) < DBL_EPSILON; }
+
+inline __device__ __host__ bool IsOne(double val)
+{
+    return fabs(val - static_cast<double>(1.0) < DBL_EPSILON);
+}
+
+template <typename dst_data_t, typename acc_data_t>
+inline __device__ void alphaBetaUpdate(dst_data_t* __restrict__ p_array,
+                                       const acc_data_t value,
+                                       double alpha,
+                                       double beta,
+                                       size_t index)
+{
+    if(IsOne(alpha) && IsZero(beta))
+    {
+        p_array[index] = cast_to<acc_data_t, dst_data_t>(value);
+        return;
+    }
+
+    acc_data_t val_alpha_beta =
```

Comment:
This is wrong. Imagine if `acc_data_t` is `int32_t` and we're casing `double` to `int32_t`. The correct way is to cast everything to `double` and then to `dst_data_t` in the second line.